### PR TITLE
Add Github workflow

### PR DESCRIPTION
**Pull request by chkn**
Adds a Github workflow to block PR merging when a label prefixed with `blocked:` is added.

This will be used for taking the API spec changes from the monolith and blocking the docs from shipping until the API changes do.

### CODE REVIEW

**Comment by chkn** [comment]
@rwdaigle I don't have access to the repo settings for this repo.. could you setup `main` as a protected branch and block merging PRs if the action added in this PR fails? (not sure if this PR needs to be merged first)

CleanShot 2024-07-31 at 17 18 57@2x

**Review by rwdaigle — APPROVED** [review]
LGTM. Branch protection added (we'll see if I've blocked myself by doing this before merging this PR):

image

**Pull request merged**
